tests/test_agent_manager.py

```diff
@@ -1272,3 +1272,31 @@ async def test_dont_start_paused_agent(server, client, environment, caplog) -> N
     assert "Started new agent with PID" not in caplog.text
     # Ensure no timeout happened
     assert "took too long to start" not in caplog.text
+
+
+async def test_auto_started_agent_log_in_debug_mode(server, environment, agent_factory, caplog, monkeypatch):
+    """
+    Test the logging of an autostarted agent
+    """
+
+    async def mock_start_agent(self) -> None:
+        await super(Agent, self).start()
+        LOGGER.debug("test log in debug level")
```

Comment:
This removes the functionality present in the `Agent.start()` method. This may cause all sorts of instabilities on the test suite. We should make sure we main that behavior.